### PR TITLE
feat: 移除已废弃的方法和逻辑 & 优化文档

### PR DESCRIPTION
**Pull request by lijinke666**
### 👀 PR includes

🎨 Enhance

- [ ] Code style optimization
- [x] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

### 📝 Description

移除一些无用 TODO 和 deprecated 的方法, 完善文档

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: s2-site/examples/custom/custom-icon/demo/custom-header-action-icon.ts [hunk 1]
```diff
@@ -1,19 +1,14 @@
-import React from 'react';
```
Comment (lijinke666) [inline s2-site/examples/custom/custom-icon/demo/custom-header-action-icon.ts:1]: 将 React 的 demo 替换成 core 版本的

File: packages/s2-core/src/interaction/event-controller.ts [hunk 2]
```diff
@@ -82,8 +82,6 @@ export class EventController {
     this.addCanvasEvent(OriginEventType.POINTER_MOVE, this.onCanvasMousemove);
     this.addCanvasEvent(OriginEventType.MOUSE_OUT, this.onCanvasMouseout);
     this.addCanvasEvent(OriginEventType.POINTER_UP, this.onCanvasMouseup);
-    // TODO: g5.0 目前没支持 dbclick 事件
-    this.addCanvasEvent(OriginEventType.DOUBLE_CLICK, this.onCanvasDoubleClick);
```
Comment (lijinke666) [inline packages/s2-core/src/interaction/event-controller.ts:86]: image

File: packages/s2-core/src/utils/text.ts [hunk 3]
```diff
@@ -55,216 +52,6 @@ export const getDisplayText = (
   return isEmptyText ? emptyPlaceholder : `${text}`;
 };
 
-/**
```
Comment (lijinke666) [inline packages/s2-core/src/utils/text.ts:58]: G 内部处理, 无法额外单独计算

**Review by wjgogogo — APPROVED** [review]

**Pull request merged**